### ISSUE TITLE
Disable `explicit_write` in test code?

### ISSUE THREAD

**Issue by smoelius** [rust-lang/rust-clippy#15780]
Would you consider a PR to disabe the [explicit_write](https://rust-lang.github.io/rust-clippy/master/index.html#explicit_write) lint in test code?

An explicit write allows you to circumvent `libtest`'s stdout/stderr capture, which you might to do to emit warnings, for example.

The below test can be used to verify this ([playground](https://play.rust-lang.org/?version=stable&mode=debug&edition=2024&gist=f29d7fbccb402e588cddca2226b5434e)):
```rs
#[test]
fn test() {
    use std::io::Write;
    writeln!(std::io::stderr(), "I am an explicit write.").unwrap();
    eprintln!("I am not an explicit write.");
}
```
